astropy/cosmology/flrw.py

```diff
@@ -141,120 +141,85 @@ def __init__(self, H0, Om0, Ode0, Tcmb0=0.0*u.K, Neff=3.04, m_nu=0.0*u.eV,
         self.m_nu = m_nu  # (reset later, this is just for unit validation)
         self.Ob0 = Ob0  # (must be after Om0)
 
-        # Derived quantities
+        # Derived quantities:
+        # Dark matter density; matter - baryons, if latter is not None.
         self._Odm0 = None if Ob0 is None else (self._Om0 - self._Ob0)
 
         # 100 km/s/Mpc * h = H0 (so h is dimensionless)
-        self._h = self._H0.value / 100.
+        self._h = self._H0.value / 100.0
         # Hubble distance
         self._hubble_distance = (const.c / self._H0).to(u.Mpc)
-        # H0 in s^-1; don't use units for speed
+        # H0 in s^-1
         H0_s = self._H0.value * H0units_to_invs
-        # Hubble time; again, avoiding units package for speed
-        self._hubble_time = u.Quantity(sec_to_Gyr / H0_s, unit=u.Gyr)
+        # Hubble time
+        self._hubble_time = sec_to_Gyr / H0_s << u.Gyr
```

Comment:
I'd tend to use parentheses here so that it is clear that the unit belongs to the whole expression.